src/relay/transforms/infer_layout_utils.h

```diff
@@ -85,6 +85,30 @@ inline Layout AdjustSubordinateFactors(const Layout& src_layout, const Layout& o
   return Layout(new_layout);
 }
 
+/*
+ * \brief An output structure to hold results from FInferCorrectLayout calls.
+ * \tparam inferred_layout An array of two elements, inferred input layouts and
+ *                         inferred output layouts.
+ * \tparam new_attrs Updated attributes consistent with inferred layouts
+ */
+class InferCorrectLayoutOutputNode : public Object {
+ public:
+  Array<Array<Layout>> inferred_layout;
```

Comment:
Can we change the inner thing to actually be a data structure instead of using Array's to represent structures/pairs?

Reply:
Thanks for the good suggestion! Updated.